cms/core/tests/test_read_only_connection.py

```diff
@@ -0,0 +1,31 @@
+from django.test import TestCase, override_settings
+from django.utils.connection import ConnectionDoesNotExist
+from wagtail_factories import ImageFactory
+
+from cms.images.models import CustomImage, Rendition
+
+
+@override_settings(IS_EXTERNAL_ENV=True)
+class ReadOnlyConnectionTestCase(TestCase):
+    def test_cannot_write_to_disallowed_table(self):
+        """Test that disallowed models cannot be written to in external env."""
+        self.assertFalse(CustomImage.objects.exists())
+
+        with self.assertRaises(ConnectionDoesNotExist):
+            ImageFactory.create()
```

Comment:
I'm not the biggest fan of this. It's misleading to have this error raised in production when it was because the writes were disabled. I would prefer if we could raise a custom exception to make the cause of the error clear.

Reply:
This would happen anyway when someone tries to run a write query anyway, this just raises a similar error. Sadly it's not viable to raise an error in `db_for_write`, as that can be obtained even for read queries, just never used.

The alternative is subclassing the connection itself and checking the queries executed, but this feels better than that.

Reply:
It looks like this may be doable with some of Django's database instrumentation APIs. I'll keep poking.

Reply:
I think the main thing we want to keep in mind is how far down the rabbit hole we're going (ie. "configure vs customise" 🙈 )

Reply:
Unfortunately, the instrumentation has no concept of models, without parsing the SQL, so this is the best way. I can use a cuustom connection to change the exception, but it's still an exception.

Reply:
Ya, it would have been a nice to have but don't want to spend long on it, or want it to become very beskpoke.

Reply:
> I can use a cuustom connection to change the exception, but it's still an exception.

That's fine, it was more to help with support, logging and alerting based on specific exceptions etc.